libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp

```diff
@@ -46,6 +46,16 @@ void NavEKF2_core::ResetVelocity(void)
             // clear the timeout flags and counters
             velTimeout = false;
             lastVelPassTime_ms = imuSampleTime_ms;
+        } else if (imuSampleTime_ms - extNavVelMeasTime_ms < 250) {
+            // use external nav data as the 2nd preference
+            // correct for antenna position
+            ext_nav_vel_elements extNavVelCorrected = extNavVelDelayed;
+            CorrectExtNavVelForSensorOffset(extNavVelCorrected.vel);
+            stateStruct.velocity = extNavVelCorrected.vel;
+            P[4][4] = P[3][3] = sq(frontend->_gpsHorizVelNoise);
```

Comment:
What if the external nav has significantly different noise/accuracy to this GPS parameter?

Reply:
If possible, it would be better for the sensor driver to set the value of accuracy and pass it through via the writeExtNavVelData function. If not available from the sensor, then the driver should set a default value.

Reply:
I agree, thank you @priseborough , I will fix this

Reply:
Hi @priseborough Should I change following codes to use ext nav vel err as well? Thank you very much https://github.com/ArduPilot/ardupilot/blob/be9e54f894faaa7562a64655b90cb1d43d15fc0a/libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp#L251-L252
and 
https://github.com/ArduPilot/ardupilot/blob/be9e54f894faaa7562a64655b90cb1d43d15fc0a/libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp#L644-L647

Reply:
Yes, whenever external nav data is being fused, the supplied accuracy should be used to set the  observation variance.

Reply:
Thank you @priseborough 
I have a question regarding gpsNEVelVarAccScale, should I create another variable like extNavVelVarAccScale? or just use it because it is a const.

Reply:
@chobitsfan, let's create a new variable even if it has the same value as the GPS one.

Reply:
I agree. I will fix it

Reply:
done now so marking as resolved.